src/backend/cuda/Array.cpp

```diff
@@ -209,28 +209,57 @@ namespace cuda
             if (node->getHeight() >= (int)getMaxJitSize()) {
                 out.eval();
             } else {
+
                 size_t alloc_bytes, alloc_buffers;
                 size_t lock_bytes, lock_buffers;
 
                 deviceMemoryInfo(&alloc_bytes, &alloc_buffers,
                                  &lock_bytes, &lock_buffers);
 
-                // Check if approaching the memory limit
-                if (lock_bytes > getMaxBytes() ||
-                    lock_buffers > getMaxBuffers()) {
+                bool isBufferLimit =
+                    lock_bytes > getMaxBytes() ||
```

Comment:
You can use [checkMemoryLimit()](https://github.com/arrayfire/arrayfire/blob/master/src/backend/common/MemoryManager.hpp#L432) for this.

Reply:
Will do.

Reply:
Can't do this because lock_bytes is used elsewhere. Fixed the other issues that were raised.